docs/apache-airflow-providers/howto/create-update-providers.rst

```diff
@@ -0,0 +1,322 @@
+      │   ├── apache-airflow/
+      │   │   └── extra-packages-ref.rst
+      │   ├── integration-logos/new_provider/
+      │   │   └── New_Provider.png
+      │   └── apache-airflow-providers-new_provider/
+      │       ├── index.rst
+      │       ├── commits.rst
+      │       ├── connections.rst
+      │       └── operators/
+      │           └── new_provider.rst
+      └── providers/
+          ├── dependencies.json
+          └── new_provider/
+              ├── provider.yaml
+              └── CHANGELOG.rst
+
+
+The ``airflow/providers/dependencies.json`` is automatically updated by pre-commit.
+
+There is a chance that your provider's name is not an common English word.
+In this case is necessary to add it to the file ``docs/spelling_wordlist.txt``. This file begin with capitalize words and
+lowercase in the second block.
+
+  .. code-block:: bash
+
+    Namespace
+    Neo4j
+    Nextdoor
+    New_Provider (new line)
+    Nones
+    NotFound
+    Nullable
+    ...
+    neo4j
+    neq
+    networkUri
+    new_provider (new line)
+    nginx
+    nobr
+    nodash
+
+Add your provider dependencies into **PROVIDER_REQUIREMENTS** variable in ``setup.py``. If your provider doesn't have
+any dependency add a empty list.
+
+  .. code-block:: python
+
+      PROVIDERS_REQUIREMENTS: Dict[str, List[str]] = {
+          ...
+          'microsoft.winrm': winrm,
+          'mongo': mongo,
+          'mysql': mysql,
+          'neo4j': neo4j,
+          'new_provider': [],
+          'odbc': odbc,
+          ...
+          }
+
+In the ``CONTRIBUTING.rst`` adds:
+
+- your provider name in the list in the **Extras** section
+- your provider dependencies in the **Provider Packages** section table, only if your provider has external dependencies.
+
+
+In the ``INSTALL`` file adds:
+- your provider to the **Extras** section list.
+
+In the ``docs/apache-airflow-providers-new_provider/connections.rst``:
+
+- add information how to connect to your providers.
+
+In the ``docs/apache-airflow-providers-new_provider/operators/new_provider.rst``:
+
+- add information how to use the Operator. It's important to add examples and additional information if your Operator has extra-parameters.
+
+  .. code-block:: RST
+
+      .. _howto/operator:NewProviderOperator:
+
+      NewProviderOperator
+      ===================
+
+      Use the :class:`~airflow.providers.new_provider.operators.NewProviderOperator` to do something
+      amazing with Airflow!
+
+      Using the Operator
+      ^^^^^^^^^^^^^^^^^^
+
+      The NewProviderOperator requires a ``connection_id`` and this other awesome parameter.
+      You can see an example below:
+
+      .. exampleinclude:: /../../airflow/providers/new_provider/example_dags/example_new_provider.py
+          :language: python
+          :start-after: [START howto_operator_new_provider]
+          :end-before: [END howto_operator_new_provider]
+
+
+In the ``docs/apache-airflow-providers-new_provider/commits.rst``:
```

Comment:
This is not needed . It will be added automatically when providers gets released.